lib/Cro/HTTP/Client/CookieJar.pm6

```diff
@@ -49,6 +49,32 @@ monitor Cro::HTTP::Client::CookieJar {
         False
     }
 
+    method !checkCookieAge(Cro::HTTP::Cookie $cookie) {
+      my ($persist, $expiry);
+      if $cookie.max-age.defined {
+          $persist = True;
+          $expiry = DateTime.now.later(seconds => $cookie.max-age)
+      }
+      elsif !$cookie.max-age.defined && $cookie.expires {
+          $persist = True;
+          $expiry = $cookie.expires;
+      } else {
+          $persist = False;
+          $expiry = DateTime.now.later(years => 10);
+      }
+      ($persist, $expiry);
+    }
+
+    method add-cookie(Cro::HTTP::Cookie $cookie) {
+      my $state = CookieState.new(
+        creation-time => DateTime.now,
+        last-access-time => DateTime.now
+      );
+
+      ($state.persistent, $state.expiry-time) = self!checkCookieAge($cookie);
+      @!cookies.push: $state.clone(cookie => $cookie.clone);
```

Comment:
Why `clone` the state here? We just created it above.

Reply:
No real reason. I was following what was done in the code where cookies were added from the request. It can be removed. I will update and recommit.

Reply:
@Altai-man Hmm, I don't see why there's a `clone` of `$state` in the existing code; am I missing something or is it redundant there too?

Reply:
It likely was meant to populate cookie attribute, but given it's marked as `rw` I don't see it as needed anymore. Tests pass without that clone.
Can be changed to
```
$state.cookie = $_.clone(:$domain, :$path);
@!cookies.push: $state;
```
or the like.

Reply:
These changes have been made.